test failures with Python 3.9 and the experimental Debian package

> FAILED (SKIP=6, errors=9, failures=1)
```
======================================================================
ERROR: Failure: TypeError (unsupported operand type(s) for *: 'GreaterThan' and 'Add')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 416, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.9/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.9/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/design.py", line 20, in <module>
    from .hrf import glover
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 123, in <module>
    _gexpr = gamma_expr(5.4, 5.2) - 0.35 * gamma_expr(10.8, 7.35)
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 106, in gamma_expr
    coef * ((T >= 0) * (T+1.0e-14))**(shape-1)
TypeError: unsupported operand type(s) for *: 'GreaterThan' and 'Add'

======================================================================
ERROR: Failure: TypeError (unsupported operand type(s) for *: 'GreaterThan' and 'Add')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 416, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.9/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.9/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/fmristat/api.py", line 3, in <module>
    from .hrf import canonical as delay_hrf
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/fmristat/hrf.py", line 21, in <module>
    from .. import hrf
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 123, in <module>
    _gexpr = gamma_expr(5.4, 5.2) - 0.35 * gamma_expr(10.8, 7.35)
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 106, in gamma_expr
    coef * ((T >= 0) * (T+1.0e-14))**(shape-1)
TypeError: unsupported operand type(s) for *: 'GreaterThan' and 'Add'

======================================================================
ERROR: Failure: TypeError (unsupported operand type(s) for *: 'GreaterThan' and 'Add')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 416, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.9/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.9/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/fmristat/hrf.py", line 21, in <module>
    from .. import hrf
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 123, in <module>
    _gexpr = gamma_expr(5.4, 5.2) - 0.35 * gamma_expr(10.8, 7.35)
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 106, in gamma_expr
    coef * ((T >= 0) * (T+1.0e-14))**(shape-1)
TypeError: unsupported operand type(s) for *: 'GreaterThan' and 'Add'

======================================================================
ERROR: Failure: TypeError (unsupported operand type(s) for *: 'GreaterThan' and 'Add')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 416, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.9/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.9/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/fmristat/tests/test_FIAC.py", line 15, in <module>
    from ... import utils, hrf, design
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 123, in <module>
    _gexpr = gamma_expr(5.4, 5.2) - 0.35 * gamma_expr(10.8, 7.35)
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 106, in gamma_expr
    coef * ((T >= 0) * (T+1.0e-14))**(shape-1)
TypeError: unsupported operand type(s) for *: 'GreaterThan' and 'Add'

======================================================================
ERROR: Failure: TypeError (unsupported operand type(s) for *: 'GreaterThan' and 'Add')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 416, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.9/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.9/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/fmristat/tests/test_hrf.py", line 11, in <module>
    from ... import hrf
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 123, in <module>
    _gexpr = gamma_expr(5.4, 5.2) - 0.35 * gamma_expr(10.8, 7.35)
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 106, in gamma_expr
    coef * ((T >= 0) * (T+1.0e-14))**(shape-1)
TypeError: unsupported operand type(s) for *: 'GreaterThan' and 'Add'

======================================================================
ERROR: Failure: TypeError (unsupported operand type(s) for *: 'GreaterThan' and 'Add')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 416, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.9/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.9/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 123, in <module>
    _gexpr = gamma_expr(5.4, 5.2) - 0.35 * gamma_expr(10.8, 7.35)
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 106, in gamma_expr
    coef * ((T >= 0) * (T+1.0e-14))**(shape-1)
TypeError: unsupported operand type(s) for *: 'GreaterThan' and 'Add'

======================================================================
ERROR: Failure: TypeError (unsupported operand type(s) for *: 'GreaterThan' and 'Add')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 416, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.9/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.9/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/tests/test_design.py", line 8, in <module>
    from ..design import (event_design, block_design, stack2designs, stack_designs,
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/design.py", line 20, in <module>
    from .hrf import glover
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 123, in <module>
    _gexpr = gamma_expr(5.4, 5.2) - 0.35 * gamma_expr(10.8, 7.35)
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 106, in gamma_expr
    coef * ((T >= 0) * (T+1.0e-14))**(shape-1)
TypeError: unsupported operand type(s) for *: 'GreaterThan' and 'Add'

======================================================================
ERROR: Failure: TypeError (unsupported operand type(s) for *: 'GreaterThan' and 'Add')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 416, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.9/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.9/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/tests/test_hrf.py", line 12, in <module>
    from ..hrf import (
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 123, in <module>
    _gexpr = gamma_expr(5.4, 5.2) - 0.35 * gamma_expr(10.8, 7.35)
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 106, in gamma_expr
    coef * ((T >= 0) * (T+1.0e-14))**(shape-1)
TypeError: unsupported operand type(s) for *: 'GreaterThan' and 'Add'

======================================================================
ERROR: Failure: TypeError (unsupported operand type(s) for *: 'GreaterThan' and 'Add')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/lib/python3/dist-packages/nose/loader.py", line 416, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python3/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/lib/python3.9/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.9/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/tests/test_utils.py", line 30, in <module>
    from .. import hrf
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 123, in <module>
    _gexpr = gamma_expr(5.4, 5.2) - 0.35 * gamma_expr(10.8, 7.35)
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/hrf.py", line 106, in gamma_expr
    coef * ((T >= 0) * (T+1.0e-14))**(shape-1)
TypeError: unsupported operand type(s) for *: 'GreaterThan' and 'Add'

======================================================================
FAIL: Doctest: nipy.modalities.fmri.utils.convolve_functions
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.9/doctest.py", line 2204, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for nipy.modalities.fmri.utils.convolve_functions
  File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/utils.py", line 494, in convolve_functions

----------------------------------------------------------------------
File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/utils.py", line 533, in nipy.modalities.fmri.utils.convolve_functions
Failed example:
    f1 = (t > 0) * (t < 1)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.9/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest nipy.modalities.fmri.utils.convolve_functions[2]>", line 1, in <module>
        f1 = (t > 0) * (t < 1)
    TypeError: unsupported operand type(s) for *: 'StrictGreaterThan' and 'StrictLessThan'
----------------------------------------------------------------------
File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/utils.py", line 538, in nipy.modalities.fmri.utils.convolve_functions
Failed example:
    tri = convolve_functions(f1, f1, [0, 2], [0, 2], 1.0e-3, name='conv')
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.9/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest nipy.modalities.fmri.utils.convolve_functions[3]>", line 1, in <module>
        tri = convolve_functions(f1, f1, [0, 2], [0, 2], 1.0e-3, name='conv')
    NameError: name 'f1' is not defined
----------------------------------------------------------------------
File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/utils.py", line 542, in nipy.modalities.fmri.utils.convolve_functions
Failed example:
    print(tri)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.9/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest nipy.modalities.fmri.utils.convolve_functions[4]>", line 1, in <module>
        print(tri)
    NameError: name 'tri' is not defined
----------------------------------------------------------------------
File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/utils.py", line 547, in nipy.modalities.fmri.utils.convolve_functions
Failed example:
    ftri = lambdify(t, tri)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.9/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest nipy.modalities.fmri.utils.convolve_functions[5]>", line 1, in <module>
        ftri = lambdify(t, tri)
    NameError: name 'tri' is not defined
----------------------------------------------------------------------
File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/utils.py", line 549, in nipy.modalities.fmri.utils.convolve_functions
Failed example:
    y = ftri(x)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.9/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest nipy.modalities.fmri.utils.convolve_functions[7]>", line 1, in <module>
        y = ftri(x)
    NameError: name 'ftri' is not defined
----------------------------------------------------------------------
File "/build/nipy-0.4.3~rc1/.pybuild/cpython3_3.9_nipy/build/nipy/modalities/fmri/utils.py", line 552, in nipy.modalities.fmri.utils.convolve_functions
Failed example:
    x[np.argmax(y)]
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.9/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest nipy.modalities.fmri.utils.convolve_functions[8]>", line 1, in <module>
        x[np.argmax(y)]
    NameError: name 'y' is not defined


----------------------------------------------------------------------
```

--- a/nipy/modalities/fmri/hrf.py
+++ b/nipy/modalities/fmri/hrf.py
@@ -103,7 +103,8 @@ def gamma_params(peak_location, peak_fwhm):
 def gamma_expr(peak_location, peak_fwhm):
     shape, scale, coef = gamma_params(peak_location, peak_fwhm)
     return (
-        coef * ((T >= 0) * (T+1.0e-14))**(shape-1)
+        coef
+        * sympy.Piecewise((T + 1e-14, T >= 0), (0, True))**(shape-1)
         * sympy.exp(-(T+1.0e-14)/scale)
         )
 
@@ -140,7 +141,7 @@ dglover = implemented_function('dglover', dglovert)
 del(_gexpr); del(_dpos); del(_dgexpr)
 
 # AFNI's HRF
-_aexpr = ((T >= 0) * T)**8.6 * sympy.exp(-T/0.547)
+_aexpr = sympy.Piecewise((T, T >= 0), (0, True))**8.6 * sympy.exp(-T/0.547)
 _aexpr = _aexpr / _get_sym_int(_aexpr)
 # Numerical function
 afnit = lambdify_t(_aexpr)

--- a/nipy/modalities/fmri/tests/test_utils.py
+++ b/nipy/modalities/fmri/tests/test_utils.py
@@ -170,8 +170,8 @@ def numerical_convolve(func1, func2, interval, dt):
 
 def test_convolve_functions():
     # replicate convolution
-    # This is a square wave on [0,1]
-    f1 = (t > 0) * (t < 1)
+    # This is a square wave on (0,1)
+    f1 = sympy.Piecewise((0, t <= 0), (1, t < 1), (0, True))
     # ff1 is the numerical implementation of same
     ff1 = lambdify(t, f1)
     # Time delta
@@ -205,7 +205,7 @@ def test_convolve_functions():
             y = ftri(time)
             assert_array_almost_equal(value, y)
         # offset square wave by 1 - offset triangle by 1
-        f2 = (t > 1) * (t < 2)
+        f2 = sympy.Piecewise((0, t <= 1), (1, t < 2), (0, True))
         tri = cfunc(f1, f2, [0, 3], [0, 3], dt)
         ftri = lambdify(t, tri)
         o1_time = np.arange(0, 3, dt)
@@ -221,7 +221,7 @@ def test_convolve_functions():
         o2_time = np.arange(0, 4, dt)
         assert_array_almost_equal(ftri(o2_time), np.r_[z1s, z1s, value])
         # offset by -0.5 - offset triangle by -0.5
-        f3 = (t > -0.5) * (t < 0.5)
+        f3 = sympy.Piecewise((0, t <= -0.5), (1, t < 0.5), (0, True))
         tri = cfunc(f1, f3, [0, 2], [-0.5, 1.5], dt)
         ftri = lambdify(t, tri)
         o1_time = np.arange(-0.5, 1.5, dt)

--- a/nipy/modalities/fmri/utils.py
+++ b/nipy/modalities/fmri/utils.py
@@ -528,9 +528,9 @@ def convolve_functions(f, g, f_interval, g_interval, dt,
     >>> from nipy.algorithms.statistics.formula.formulae import Term
     >>> t = Term('t')
 
-    This is a square wave on [0,1]
+    This is a square wave on (0,1)
 
-    >>> f1 = (t > 0) * (t < 1)
+    >>> f1 = sympy.Piecewise((0, t <= 0), (1, t < 1), (0, True))
 
     The convolution of ``f1`` with itself is a triangular wave on [0, 2],
     peaking at 1 with height 1

